content/docs/learning-guide/tutorials/rendering/vegetation-bending-tutorial.md

```diff
@@ -0,0 +1,704 @@
+
+1. Open `{your-project-path}\Materials\Types\VegetationBending_Common.azsli`.
+1. At the bottom of the file, add:
+
+   ```
+   option bool o_color_isBound;
+   ```
+   We place this variable in the common file because we will be using it in all the passes.
+1. Open `{your-project-path}\Materials\Types\VegetationBending_ForwardPass.azsli`. Inside `struct VegetationVSInput`, add another field:
+
+   ```
+   float4 m_optional_color : COLOR0;
+   ```
+   If the material's vertices are colored, `m_optional_color` will be set at runtime if it's available. Then, if `m_optional_color` is available, a soft naming convention will set `o_color_isBound` to true, which we can use later to determine if we want to perform the bending or not. Note that this soft naming convention is a very specific sub-feature of shader options that are set based on the presence or absence of an optional vertex stream, as opposed to shader options set based on material properties.
+   
+   All of the fields are indicated by [HLSL semantics](https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-semantics#vertex-shader-semantics). The engine processes the semantics and updates the fields accordingly.
+1. Inside the function `VegetationBending_ForwardPassVS`, encase the offset code with a conditional using the option boolean:
+
+   ```
+   if (o_color_isBound) {
+      worldPosition.x += MaterialSrg::m_xOffset;
+      worldPosition.y += MaterialSrg::m_yOffset;
+   }
+   ```
+1. Repeat steps 3 and 4 with the depth pass in `{your-project-path}\Materials\Types\VegetationBending_DepthPass.azsli`.
+1. Save both files and open your level in the **Editor** from the previous steps. You should see that the tree entity is offset, but the shader ball (which doesn't have a vertex color stream) is not!
+
+{{< image-width src="/images/atom-guide/vegetation-bending-tutorial/optionoffset.png" width="100%" alt="Option offset added." >}}
+
+### Delete previous offset code
+Now that we have our tree and tested the shader options, let's delete the previous offset code so that we can start fresh with vegetation bending. That is, delete the following:
+* The code added to adjust the position in the vertex shaders of both the forward pass and depth pass
+* The `m_xOffset` and `m_yOffset` variable declarations in the `VegetationBending_Common.azsli` file in the `MaterialSrg`
+* The `xOffset` and `yOffset` properties in the `VegetationBendingPropertyGroup.json` file, but keep the rest of the file and its connection in `VegetationBending.materialtype`; we will need it for later!
+
+Make sure you are keeping the declarations for `m_optional_color` and `o_color_isBound`.
+
+### Create materials for the tree
+Let's add some textures to make our tree look more realistic! For the tree, we need 3 materials: one for the trunk, one for the branches, and one for the leaves.
+
```

Comment:
For the steps in this section, I suggest revising to better streamline the instructions. It's best to put the goal of the step near the beginning. 
You can refer back to my comments for **Add a tree** section for examples.

Reply:
Made some changes to this section with putting the goal at the beginning of the step and shortening the actions!